src/tests/JIT/Regression/JitBlue/Runtime_100437/Runtime_100437.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.Loader;
+using Xunit;
+
+public class Runtime_100437
+{
+    [Fact]
+    public static int TestCollectibleEmptyArrayNotInFrozenHeap()
+    {
+        string assemblyPath = typeof(Runtime_100437).Assembly.Location;
+
+        // Skip this test for single file
+        if (string.IsNullOrEmpty(assemblyPath))
+            return 100;
+
+        WeakReference[] wrs = new WeakReference[10];
+        for (int i = 0; i < wrs.Length; i++)
+        {
+            var alc = new MyAssemblyLoadContext();
+            var a = alc.LoadFromAssemblyPath(assemblyPath);
+            wrs[i] = (WeakReference)a.GetType(nameof(Runtime_100437)).GetMethod(nameof(Work)).Invoke(null, null);
+            GC.Collect();
+        }
+
+        int result = 0;
+        foreach (var wr in wrs)
+        {
+            // This is testing that the empty array from Work(), if collected, should not have been allocated on the Frozen heap
+            // otherwise it will result in a random crash.
+            result += wr.Target?.ToString()?.GetHashCode() ?? 0;
+        }
+        return (result & 100) | 100;
+    }
+
+    [Fact]
+    public static int TestCollectibleReadOnlyStaticWithGeneric()
+    {
+        string assemblyPath = typeof(Runtime_100437).Assembly.Location;
+
+        // Skip this test for single file
+        if (string.IsNullOrEmpty(assemblyPath))
+            return 100;
+
+        WeakReference[] wrs = new WeakReference[10];
+        for (int i = 0; i < wrs.Length; i++)
+        {
+            var alc = new MyAssemblyLoadContext();
+            var a = alc.LoadFromAssemblyPath(assemblyPath);
+            wrs[i] = (WeakReference)a.GetType(nameof(Runtime_100437)).GetMethod(nameof(WorkForGenericCase)).Invoke(null, null);
+            GC.Collect();
+        }
+
+        int result = 0;
+        foreach (var wr in wrs)
+        {
+            result += wr.Target?.ToString()?.GetHashCode() ?? 0;
+        }
+        return (result & 100) | 100;
+    }
```

Comment:
```suggestion
    public static int TestCollectibleNonGenericReadOnlyStaticInSharedGenericType()
    {
        string assemblyPath = typeof(Runtime_100437).Assembly.Location;

        // Skip this test for single file
        if (string.IsNullOrEmpty(assemblyPath))
            return 100;

        WeakReference wr = CreateReadOnlyStaticWeakReference();

        for (int i = 0; i < 10; i++)
        {
            GC.Collect();
            GC.WaitForPendingFinalizers();
            if (wr.Target == null)
                return 100;
        }

        Console.WriteLine("Test Failed")
        return 1;
        
        [MethodImpl(MethodImplOptions.NoInlining)]
        WeakReference CreateReadOnlyStaticWeakReference()
        {
                AssemblyLoadContext alc = new MyAssemblyLoadContext();
                Assembly a = alc.LoadFromAssemblyPath(assemblyPath);
                return (WeakReference)a.GetType(nameof(Runtime_100437)).GetMethod(nameof(WorkForGenericCase)).Invoke(null, null);
            }
        }
    }
```
This variant needs to check the static disappears after the ALC is collected.

Reply:
Actually, this is not testing the interesting path.